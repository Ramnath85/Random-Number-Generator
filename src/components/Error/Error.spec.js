import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Error from "./Error";

configure({ adapter: new Adapter() });

describe('Test Error component', () => {
  let spy;
  let shallowComponent;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      error: false,
      message: ''
    };
    shallowComponent = shallow(<Error {...defaultProps} />);
    spy = shallowComponent.instance();
  });

  test('should render the error component', () => {
    expect(shallowComponent).toMatchSnapshot();
    expect(shallowComponent.length).toEqual(1);
  });
});