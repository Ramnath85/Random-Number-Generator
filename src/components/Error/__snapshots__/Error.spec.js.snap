// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Error component should render the error component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error
    error={false}
    message=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": false,
    },
    "ref": null,
    "rendered": false,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": false,
      },
      "ref": null,
      "rendered": false,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
